src/App.tsx

```diff
@@ -1,14 +1,71 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState, useCallback } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
-
+import { getTodos } from './api';
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+// import { event } from 'cypress/types/jquery';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [status, setStatus] = useState('');
+  const [query, setQuery] = useState('');
+
+  useEffect(() => {
+    setIsLoading(true);
+
+    getTodos()
+      .then(todosFromServer => {
+        setTodos(todosFromServer);
+      })
+      .finally(() => setIsLoading(false));
+  }, []);
+
+  const handleStatus = useCallback((selectedStatus: string) => {
+    setStatus(selectedStatus);
+  }, []);
+
+  const handleQuery = useCallback((search: string) => {
+    setQuery(search);
+  }, []);
+
+  const handleTodo = (todo: Todo) => {
+    setSelectedTodo(todo);
+  };
+
+  const clearSelectedTodo = () => {
+    setSelectedTodo(null);
+  };
+
+  const filterTodos = useCallback((search: string, todoStatus: string) => {
+    if (todoStatus === 'completed') {
+      return todos.filter(todo => (
+        todo.completed && todo.title.toLowerCase().includes(search.toLowerCase())
+      ));
+    }
+
+    if (todoStatus === 'active') {
+      return todos.filter(todo => (
+        !todo.completed && todo.title.toLowerCase().includes(search.toLowerCase())
+      ));
+    }
+
+    return todos.filter(todo => (
+      todo.title.toLowerCase().includes(search.toLowerCase())));
+  }, [query, status, todos]);
```

Comment:
you have a limited number of statuses, so it would be better to use `switch`, and also try not to write the same code several times :)

```suggestion
  const filterTodos = todos.filter(todo => {
    const normalizedTodo
    = todo.title.toLowerCase().includes(searchQuery.toLowerCase());

    switch (todoStatus) {
      case 'all':
        return normalizedTodo;
      case 'active':
        return normalizedTodo && !todo.completed;
      case 'completed':
        return normalizedTodo && todo.completed;
      default:
        throw new Error('...');
    }
  });
```